karoo_gp/base_class.py

```diff
@@ -675,32 +447,162 @@ def fx_data_params_write_json(self):
         if len(self.population.fittest_dict) == 0:
             final_dict['outcome'] = 'FAILURE'
         else:
-            tree = self.fx_eval_fittest()
+            tree = self.population.fittest()
             fittest_tree = dict(
                 id=tree.id,
                 expression=tree.expression,
             )
-            result = tree.result
-            score = dict(fitness=result['fitness'])
-            if self.kernel == 'c':
-                score['classification_report'] = skm.classification_report(
-                    result['solution'], result['pred_labels'][0],
-                    output_dict=True,
-                )
-                score['confusion_matrix'] = skm.confusion_matrix(
-                    result['solution'], result['pred_labels'][0]
-                ).tolist()
-
-            elif self.kernel == 'r':
-                MSE = skm.mean_squared_error(result['result'], result['solution'])
-                score['mean_squared_error'] = float(MSE)
-
             final_dict = dict(
                 **final_dict,
                 outcome='SUCCESS',
                 fittest_tree=fittest_tree,
-                score=score,
+                score=self.score(),
             )
-
-        with open(self.path + 'results.json', 'w') as f:
+        with open(self.loader.path + 'results.json', 'w') as f:
             json.dump(final_dict, f, indent=4)
+
+# _____________________________________________________________________________
+# KERNEL IMPLEMENTATIONS
+
+class Regressor_GP(Base_GP):
```

Comment:
You should drop the `_` from all classes' names:
```suggestion
class RegressorGP(BaseGP):
```